src/lib/storage/table.cpp

```diff
@@ -96,12 +100,42 @@ std::shared_ptr<const Chunk> Table::get_chunk(ChunkID chunk_id) const {
   return _chunks.at(chunk_id);
 }
 
-// GCOVR_EXCL_START
-void Table::compress_chunk(const ChunkID chunk_id) {
-  // Implementation goes here
-  Fail("Implementation is missing.");
+void compress_segment(const std::shared_ptr<AbstractSegment> segment,
+                      std::vector<std::shared_ptr<AbstractSegment>>& compressed_segments, ColumnID segment_index,
+                      std::string type) {
+  resolve_data_type(type, [&](const auto data_type_t) {
+    using ColumnDataType = typename decltype(data_type_t)::type;
+    auto compressed_segment = std::make_shared<DictionarySegment<ColumnDataType>>(segment);
+    compressed_segments[segment_index] = compressed_segment;
+  });
 }
 
-// GCOVR_EXCL_STOP
+void Table::compress_chunk(const ChunkID chunk_id) {
+  auto old_chunk = get_chunk(chunk_id);
+  auto segment_count = old_chunk->column_count();
+  auto new_chunk = std::make_shared<Chunk>();
+  auto compressed_segments = std::vector<std::shared_ptr<AbstractSegment>>(segment_count);
+
+  auto threads = std::vector<std::thread>();
+  for (auto segment_index = ColumnID{}; segment_index < segment_count; ++segment_index) {
+    auto type = this->column_type(segment_index);
+    auto old_segment = old_chunk->get_segment(segment_index);
+    auto worker = std::thread(compress_segment, old_segment, std::ref(compressed_segments), segment_index, type);
+    threads.push_back(std::move(worker));
+  }
+  // threads join
+  for (auto thread_index = size_t{0}; thread_index < segment_count; ++thread_index) {
+    threads[thread_index].join();
+  }
+
+  for (auto segment_index = ColumnID{}; segment_index < segment_count; ++segment_index) {
+    new_chunk->add_segment(compressed_segments[segment_index]);
+  }
```

Comment:
Here the same as above
```suggestion
  for (const auto& segment : compressed_segments) {
    new_chunk->add_segment(segment);
  }
```